### PR TITLE
Add support for communication with cli

### PR DESCRIPTION
**Pull request by cooligus**

### CODE REVIEW

**Review by maxidragon — APPROVED** [review]
Please wait for Bartek's review before merging

File: src-tauri/src/main.rs
Comment (BaderBC) [inline src-tauri/src/main.rs:6]:
```suggestion

#[cfg(target_os = "linux")]
```

We should add it in order to allow compilation only for linux

File: src-tauri/src/main.rs
Comment (BaderBC) [inline src-tauri/src/main.rs]:
```suggestion
```

It isn't necessary because we support only linux

File: src-tauri/src/spito_helper.rs
Comment (BaderBC) [inline src-tauri/src/spito_helper.rs]:
```suggestion
pub fn start_spito_cli(ruleset: &str, rule: &str) -> Result<(), String> {
```

File: src-tauri/src/spito_helper.rs
Comment (BaderBC) [inline src-tauri/src/spito_helper.rs]:
```suggestion
pub fn start_spito_server(app: AppHandle) -> Result<(), String> {
```

File: src-tauri/src/main.rs
Comment (BaderBC) [inline src-tauri/src/main.rs]:
```suggestion
        .expect("error while running spito-gui application");
```

File: src-tauri/src/spito_helper.rs
Comment (BaderBC) [inline src-tauri/src/spito_helper.rs]: Maybe some more meaningful like e.g. `DbusPayload`

**Review by BaderBC — CHANGES_REQUESTED** [review]

**Review by BaderBC — CHANGES_REQUESTED** [review]
Don't listen for `Echo` and add support for `api.info.*`

**Comment by cooligus** [comment]
> Don't listen for `Echo` and add support for `api.info.*`

`Echo` is much better than creating 5 function that does completely same stuff

**Comment by BaderBC** [comment]
> > Don't listen for `Echo` and add support for `api.info.*`
> 
> 
> 
> `Echo` is much better than creating 5 function that does completely same stuff

When a rule calls `api.info.log` it should push other part of gui than other methods. e.g. `api.info.error` could be shown as a red popup, while `api.info.log` could be hidden somewhere

**Comment by cooligus** [comment]
> > > Don't listen for `Echo` and add support for `api.info.*`
> > 
> > 
> > `Echo` is much better than creating 5 function that does completely same stuff
> 
> When a rule calls `api.info.log` it should push other part of gui than other methods. e.g. `api.info.error` could be shown as a red popup, while `api.info.log` could be hidden somewhere

whether it's error or something else specifies first arguemnt of `echo`

File: src-tauri/src/spito_helper.rs
Comment (BaderBC) [inline src-tauri/src/spito_helper.rs]: Make it camel case when serialized (you can use serde decorator for this) and consider using full name

File: src-tauri/src/spito_helper.rs
Comment (BaderBC) [inline src-tauri/src/spito_helper.rs]: What about renaming `Echo` to `Info` or `api.info`

File: src-tauri/src/spito_helper.rs
Comment (cooligus) [reply src-tauri/src/spito_helper.rs:22]: RustRover says it should be snake case

**Review by BaderBC — APPROVED** [review]
LGTM!

**Pull request merged**
